wtxmgr/doc.go

```diff
@@ -0,0 +1,97 @@
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+// TODO: This package doco needs to be updated for the new wtxmgr API.
+//
+// Package wtxmgr provides an implementation of a transaction store for a
+// bitcoin wallet.  Its primary purpose is to save transactions with outputs
+// spendable with wallet keys and transactions that are signed by wallet keys in
+// memory, handle spend tracking for newly-inserted transactions, and report the
+// spendable balance from each unspent transaction output. It uses walletdb as
+// the backend for storing the serialized transaction objects in buckets.
+//
+// Transaction outputs which are spendable by wallet keys are called credits
+// (because they credit to a wallet's total spendable balance) and are modeled
+// using the Credit structure.  Transaction inputs which spend
+// previously-inserted credits are called debits (because they debit from the
+// wallet's spendable balance) and are modeled using the Debit structure.
+//
+// Besides just saving transactions, bidirectional spend tracking is also
+// performed on each credit and debit.  Unlike packages such as btcdb, which
+// only mark whether a transaction output is spent or unspent, this package
+// always records which transaction is responsible for debiting (spending) any
+// credit.  Each debit also points back to the transaction credit it spends.
+//
+// A significant amount of internal bookkeeping is used to improve the
+// performance of inserting transactions and querying commonly-needed data.
+// Most notably, all unspent credits may be iterated over without including (and
+// ignoring) spent credits.  Another trick is to record the total spendable
+// amount delta as a result of all transactions within a block, which is the
+// total value of all credits (both spent and unspent) minus the total value
+// debited from previous transactions, for every transaction in that block.
+// This allows for the calculation of a wallet's balance for any arbitrary
+// number of confirmations without needing to iterate over every unspent credit.
+//
+// Finally, this package records transaction insertion history (such as the date
+// a transaction was first received) and is able to create the JSON reply
+// structure for RPC calls such as listtransactions for any saved transaction.
+//
+// To use the transaction store, a transaction must be first inserted with
+// InsertTx.  After an insert, credits and debits may be attached to the
+// returned transaction record using the AddCredit and AddDebits methods.
+//
+// Example use:
```

Comment:
I'd suggest creating runnable example via an `example_test.go` file (see [btcjson's examples](https://github.com/btcsuite/btcd/blob/master/btcjson/v2/btcjson/example_test.go) for reference).  Examples in doc.go almost always end up out-of-date and non-functional over time.  By creating them as runnable examples, they show up under the examples category on godoc and also have the benefit of being executed when go test runs along with comparing their output to the expected output.